### ISSUE TITLE
Intermittent Lighting Change

### ISSUE THREAD

**Issue by SRSaunders** [RobertBeckebans/RBDOOM-3-BFG#889]
See the following.  Very reproducible just past the kitchen on the way to marine HQ.

https://github.com/RobertBeckebans/RBDOOM-3-BFG/assets/82544213/b5e771e6-ecf5-4b62-a650-e698120f7e0b

**Comment by RobertBeckebans** [comment]
Yeah this happens when the interpolation triangle flips between 4 nearest env_probes.
The current implementation is very simple where I just look for the 3 nearest env_probes and trilerp between them. I never got around to do a full implementation of #497.

I wanted to it similar like in Rage however Rage does it interpolate over time so it is not so sudden.

**Comment by RobertBeckebans** [comment]
I haven't addressed this issue yet because it's not apparent if you shoot monsters.

**Comment by RobertBeckebans** [comment]
Happy new year @SRSaunders !

I don't have a good solution for the flickering of the specular lighting yet but I could reduce a lot the flickering of the diffuse lightgrid lighting because there has been a bug for 3 years.

The bugfix from #965 reduces flickering from the lightgrids.

**Comment by SRSaunders** [comment]
Thanks @RobertBeckebans, same Happy New Year wishes to you as well!

I was away on vacation for a few weeks but am back now.  Some specular flickering is still present, but your diffuse lightgrid fix definitely helps, and also seems to improve the issue I reported in #939.
